src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -0,0 +1,467 @@
+                );
+        }
+
+        [Fact]
+        public void UnexpectedAccessor_02()
+        {
+            // implementation unexpected getter
+            var source = """
+                partial class C
+                {
+                    partial int P { set; }
+                    partial int P { get => throw null!; set { } }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9305: Property accessor 'C.P.get' does not implement any accessor declared on the definition part
+                //     partial int P { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyUnexpectedAccessor, "P").WithArguments("C.P.get").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void UnexpectedAccessor_03()
+        {
+            // implementation unexpected init
+            var source = """
+                partial class C
+                {
+                    partial int P { get; }
+                    partial int P { get => throw null!; init { } }
+                }
+                """;
+            var comp = CreateCompilation([source, IsExternalInitTypeDefinition]);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9305: Property accessor 'C.P.init' does not implement any accessor declared on the definition part
+                //     partial int P { get => throw null!; init { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyUnexpectedAccessor, "P").WithArguments("C.P.init").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void AccessorKind_01()
+        {
+            // set vs init
+            var source = """
+                partial class C
+                {
+                    partial int P { get; set; }
+                    partial int P { get => throw null!; init { } }
+                }
+                """;
+            // PROTOTYPE(partial-properties): give an error diagnostic for an accessor kind difference
+            var comp = CreateCompilation([source, IsExternalInitTypeDefinition]);
+            comp.VerifyEmitDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void Extern_01()
+        {
+            // PROTOTYPE(partial-properties): test that appropriate flags are set in metadata for the property accessors.
+            // See ExtendedPartialMethodsTests.Extern_Symbols as a starting point.
+            var source = """
+                partial class C
+                {
+                    partial int P { get; set; }
+                    extern partial int P { get; set; }
+                }
+                """;
+            var comp = CreateCompilation([source, IsExternalInitTypeDefinition]);
+            comp.VerifyEmitDiagnostics(
```

Comment:
Can we verify execution too? PROTOTYPE is fine too since not the most core scenario :-P

Reply:
We don't do any such verification in unit tests in e.g. `ExtendedPartialMethodsTests` because it is excessively difficult to set up a test scenario that will do something meaningful (requires deploying a native library alongside the test).